common-libs/user-management/src/CognitoIdentityService.ts

```diff
@@ -171,9 +179,9 @@ export class CognitoIdentityService {
 
   async completeLogInPasswordless(token: string, confirmationCode: string): Promise<CompleteLoginPasswordlessResponse> {
     const { Session: tokenSession, ChallengeName, ChallengeParameters } = JSON.parse(token)
-
-    const { USERNAME } = ChallengeParameters
-    const Session = tempSession[USERNAME] || tokenSession
+    //TODO: session is approx 920 character long string
+    const hashedTokenSession = sha256(tokenSession)
```

Comment:
Why do we need to hash it, instead of using session itself (probably with Map instead of a plain object)?

Reply:
session is almost 1kb long . so it is to expensive to use it as a key

Reply:
But we're removing it after successful login, and we already cache DID documents etc, I don't think having a session in memory (only until it's removed with a successful login) will cause us any troubles.
Or if it will, let's fix them later. Right now it feels like increased complexity (and potential incompatibility with the browser environment as you yourself mentioned) for the sake of premature optimization.

Reply:
Not really we have a cases when user enter a wrong OTP and never complete a flow . so it will grow 2x in memory over time.

Reply:
So in the end we're cleaning it up now in most cases.
And in the remaining cases (e.g. user entered a wrong OTP and never completed the flow) we still have a memory leak with the current approach, except that we leak 32 bytes (+C) instead of a kilobyte (+C) per abandoned session.

This is not going to solve memory leaks, only make them smaller somewhat.
And I don't see the point in increasing complexity in making these memory leaks smaller until we are certain that their size is causing issues.
Maybe even with a kilobyte per session we wouldn't have any problems.
Or maybe even with 32 bytes per session we would have serious problems.

Also we're caching DIDs currently, which are much larger, and this was not causing us any trouble so far.